File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
  let answer = "";
  let obj = {
    "R": 0,
    "T": 0,
    "C": 0,
    "F": 0,
    "J": 0,
    "M": 0,
    "A": 0,
    "N": 0,
  }
  let response = [0, 3, 2, 1, 0, 1, 2, 3];
  for (i in survey) {
    let select1 = survey[i].split("")[0];
    let select2 = survey[i].split("")[1];
    if (choices[i]<4) {
      obj[`${select1}`] += response[choices[i]];
    } else {
      obj[`${select2}`] += response[choices[i]];
    }
  }
  answer+=(obj["R"]>=obj["T"] ? "R" : "T")
  answer+=(obj["C"]>=obj["F"] ? "C" : "F")
  answer+=(obj["J"]>=obj["M"] ? "J" : "M")
  answer+=(obj["A"]>=obj["N"] ? "A" : "N")

  return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    allcategories = ['R','T','C','F','J','M','A','N']
    surveydict = {i:0 for i in allcategories}
    num = len(survey)
    for i in range(num):
        category = list(survey[i])
        choice = choices[i]
        if choice<4:
            surveydict[category[0]] += 4-choice
        elif choice>4:
            surveydict[category[1]] += choice-4
    
    result = ""
    for i in range(4):
        a = 2*i
        b = a + 1
        if surveydict[allcategories[a]] >= surveydict[allcategories[b]]:
            result+=allcategories[a]
        else:
            result+=allcategories[b]
            
    return result